ext/ManoptJuMPExt.jl

```diff
@@ -408,29 +615,70 @@ function JuMP.in_set_string(mime, set::ManifoldsBase.AbstractManifold)
 end
 
 """
-    JuMP.build_variable(::Function, func, m::ManifoldsBase.AbstractManifold)
+    JuMP.build_variable(::Function, func, manifold::ManifoldsBase.AbstractManifold)
 
 Build a `JuMP.VariablesConstrainedOnCreation` object containing variables
-and the [`Manopt.JuMP_VectorizedManifold`](@ref) in which they should belong as well as the
+and the [`ManifoldSet`](@ref) in which they should belong as well as the
 `shape` that can be used to go from the vectorized MOI representation to the
-shape of the manifold, that is, [`Manopt.JuMP_ArrayShape`](@ref).
+shape of the manifold, that is, a [`ManifoldArrayShape`](@ref).
 """
-function JuMP.build_variable(::Function, func, m::ManifoldsBase.AbstractManifold)
-    shape = _shape(m)
+function JuMP.build_variable(::Function, array, M::ManifoldsBase.AbstractManifold)
+    shape = _shape(M)
     return JuMP.VariablesConstrainedOnCreation(
-        JuMP.vectorize(func, shape), VectorizedManifold(m), shape
+        JuMP.vectorize(array, shape), ManifoldSet(M), shape
     )
 end
 
+#
+#
+# NonArrayPoints define own variable
+# TODO: Document
+struct ManifoldVariable{P<:ManifoldsBase.AbstractManifoldPoint} <: JuMP.AbstractVariable
+    p::P
+end
+# Taken / adapted from https://github.com/JuliaManifolds/Manopt.jl/pull/466#issuecomment-2862071520
+# TODO: I think I would prefer PoincareHalfPlanePoint(p) in Hyperbolic(3),
+# but it seems the in and such where not present for polynomials?
+function JuMP.build_variable(
+    _error::Function,
+    info::JuMP.VariableInfo,
+    p::ManifoldsBase.AbstractManifoldPoint;
+    extra_kwargs...,
+)
+    # cvarchecks(_error, info; extra_kwargs...) # What does this do? Necessary?
+    # _warnbounds(_error, p, info)              # What does this do?
+    return ManifoldVariable(p)
+end
+#
+#
+# TODO: Understand parameters here and document them
+function JuMP.add_variable(model::JuMP.AbstractModel, v::ManifoldVariable, name::String="")
+    # What does this do? This looks like pure magic to me.
+    #=
+    function _newvar(i)
+        vref = VariableRef(model)
```

Comment:
You can remove that, this is specific to PolyJuMP

Reply:
I removed this. But I am still not sure, for this and the other two functions I just merely guessed what might be right without understanding anything.

Do these now work? How would one define a variable to be a `PoincareHalfPlanePoint` on some `Hyperbolic(2)`?